test/connectors/CdsFileUploadConnectorSpec.scala

```diff
@@ -72,4 +78,46 @@ class CdsFileUploadConnectorSpec extends UnitSpec with BeforeAndAfterEach with I
       }
     }
   }
+
+  "CdsFileUploadConnector on getDeclarationStatus" should {
+
+    "call HttpClient" in {
+
+      val httpResponse = HttpResponse(status = OK, body = "")
+
+      when(httpClient.GET[HttpResponse](anyString())(any(), any(), any()))
+        .thenReturn(Future.successful(httpResponse))
+
+      cdsFileUploadConnector.getDeclarationStatus(MRN(CommonTestData.mrn).get)(hc).futureValue
+
+      verify(httpClient).GET[HttpResponse](anyString())(any(), any(), any())
+    }
+
+    "return value returned from HttpClient" when {
+
+      "response has Ok (200) status" in {
+
+        val httpResponse = HttpResponse(status = OK, body = "")
+
+        when(httpClient.GET[HttpResponse](anyString())(any(), any(), any()))
+          .thenReturn(Future.successful(httpResponse))
+
+        val result = cdsFileUploadConnector.getDeclarationStatus(MRN(CommonTestData.mrn).get)(hc).futureValue
+
+        result mustBe httpResponse
+      }
+
+      "response has NotFound (404) status" in {
+
+        val httpResponse = HttpResponse(status = NOT_FOUND, body = "")
+
+        when(httpClient.GET[HttpResponse](anyString())(any(), any(), any()))
+          .thenReturn(Future.successful(httpResponse))
+
+        val result = cdsFileUploadConnector.getDeclarationStatus(MRN(CommonTestData.mrn).get)(hc).futureValue
+
+        result mustBe httpResponse
+      }
```

Comment:
Should we also test that any 'other' status throws the expected exception?

Reply:
I was thinking about it, but these two are the only ones we care about. And if I were to add more statuses, there is a question "how many statuses should I test against?".
Maybe it is reasonable to add 500 and be done with it for now?